core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -766,9 +782,11 @@ private[spark] class TaskSetManager(
         logWarning(failureReason)
         None
     }
-    // always add to failed executors
-    failedExecutors.getOrElseUpdate(index, new HashMap[String, Long]()).
-      put(info.executorId, clock.getTimeMillis())
+
+    if (reason.countTowardsTaskFailures) {
```

Comment:
@kayousterhout failedExecutors itself is not relevant with this patch.
Agree about moving to within 801-if